test_util/dcos_api_session.py

```diff
@@ -89,31 +68,92 @@ def api_request(self, method, path_extension, *, scheme=None, host=None, query=N
 
 
 class DcosApiSession(ARNodeApiClientMixin, RetryCommonHttpErrorsMixin, ApiClientSession):
-    def __init__(self, dcos_url: str, masters: list, public_masters: list,
-                 slaves: list, public_slaves: list, default_os_user: str,
-                 auth_user: Optional[DcosUser]):
-        """Proxy class for DC/OS clusters.
+    def __init__(
+            self,
+            dcos_url: str,
+            masters: Optional[List[str]],
+            slaves: Optional[List[str]],
+            public_slaves: Optional[List[str]],
+            default_os_user: str,
+            auth_user: Optional[DcosUser]):
+        """Proxy class for DC/OS clusters. If any of the host lists (masters,
+        slaves, public_slaves) are provided, the wait_for_dcos function of this
+        class will wait until provisioning is complete. If these lists are not
+        provided, then there is no ground truth and the cluster will be assumed
+        the be in a completed state.
 
         Args:
             dcos_url: address for the DC/OS web UI.
             masters: list of Mesos master advertised IP addresses.
-            public_masters: list of Mesos master IP addresses routable from
-                the local host.
             slaves: list of Mesos slave/agent advertised IP addresses.
+            public_slaves: list of public Mesos slave/agent advertised IP addresses.
             default_os_user: default user that marathon/metronome will launch tasks under
             auth_user: use this user's auth for all requests
                 Note: user must be authenticated explicitly or call self.wait_for_dcos()
         """
         super().__init__(Url.from_string(dcos_url))
-        self.masters = sorted(masters)
-        self.public_masters = sorted(public_masters)
-        self.slaves = sorted(slaves)
-        self.public_slaves = sorted(public_slaves)
-        self.all_slaves = sorted(slaves + public_slaves)
+        self.master_list = masters
+        self.slave_list = slaves
+        self.public_slave_list = public_slaves
         self.default_os_user = default_os_user
         self.auth_user = auth_user
 
-        assert len(self.masters) == len(self.public_masters)
+    @staticmethod
+    def get_args_from_env():
+        """ Provides the required arguments for a unauthenticated cluster
+        """
+        masters = os.getenv('MASTER_HOSTS')
```

Comment:
The point of this PR is that you CAN have valid tests without declaring ANYTHING

The only reason we want the user to input the master list, agent list, and public agent list is so that we can verify that provisioning was 100% successful. This will still be done in all of our PRs. However, when you just have an arbitrary cluster and want to test it, you don't want to have to provide ground-truth when you just want your current cluster tested.

Anyway, I already added logging for when these params are not provided and are set by the harness: https://github.com/dcos/dcos/pull/1508/files#diff-f50a13699d4598c4a33fe8b1701335e8R137

I can bump it up from debug to warning as that might be more appropriate